src/reverse_complement.rs

```diff
@@ -58,85 +52,100 @@ impl Tables {
     fn cpl8(&self, i: u8) -> u8 {
         self.table8[i as usize]
     }
+}
 
-    /// Retrieves the complement for `i`.
-    fn cpl16(&self, i: u16) -> u16 {
-        self.table16[i as usize]
+/// Utilities for splitting chunks off of mutable slices.
+trait SplitOff {
+    fn split_off_left(&mut self, n: usize) -> Self;
+    fn split_off_right(&mut self, n: usize) -> Self;
+}
+impl<'a, T> SplitOff for &'a mut [T] {
+    /// Split the left `n` items from self and return them as a separate slice.
+    fn split_off_left(&mut self, n: usize) -> Self {
+        let n = cmp::min(self.len(), n);
+        let data = mem::replace(self, &mut []);
+        let (left, data) = data.split_at_mut(n);
+        *self = data;
+        left
+    }
+    /// Split the right `n` items from self and return them as a separate slice.
+    fn split_off_right(&mut self, n: usize) -> Self {
+        let len = self.len();
+        let n = cmp::min(len, n);
+        let data = mem::replace(self, &mut []);
+        let (data, right) = data.split_at_mut(len - n);
+        *self = data;
+        right
     }
 }
 
-/// Length of a normal line without the terminating \n.
-const LINE_LEN: usize = 60;
-const SEQUENTIAL_SIZE: usize = 1024;
+/// Length of a normal line including the terminating \n.
+const LINE_LEN: usize = 61;
+const SEQUENTIAL_SIZE: usize = 2048;
 
-/// Compute the reverse complement with the sequence split into two equal-sized slices.
-fn reverse_complement_left_right(left: &mut [u16], right: &mut [u16], tables: &Tables) {
+/// Compute the reverse complement for two contiguous chunks without line breaks.
+fn reverse_complement_chunk(left: &mut [u8], right: &mut [u8], tables: &Tables) {
```

Comment:
I tried this and it did not change performance at all.  The code is at https://github.com/mbrubeck/benchmarksgame-rs/commit/e133026d70e8fa39ff5d1b69ab9dbb2241a8fc1b.